app/data/private/key.go

```diff
@@ -0,0 +1,53 @@
+package private
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/base64"
+
+	"github.com/pkg/errors"
+	"gopkg.in/square/go-jose.v2"
+)
+
+type Key struct {
+	JWK jose.JSONWebKey
+	*rsa.PrivateKey
+}
+
+// Wrap the provided RSA private key as our internal key with canonical ID
+func NewKey(key *rsa.PrivateKey) (*Key, error) {
+	id, err := keyID(&key.PublicKey)
+	if err != nil {
+		return nil, errors.Wrap(err, "private.NewKey")
```

Comment:
authn-server currently uses `errors.Wrap` to describe the method that returned an error rather than the method wrapping said error:

```suggestion
		return nil, errors.Wrap(err, "private.keyID")
```